src/Uno.UI.RemoteControl.Server.Processors/HotReload/ServerHotReloadProcessor.cs

```diff
@@ -530,9 +552,41 @@ private async Task ProcessUpdateFile(UpdateFile message)
 					return (FileUpdateResult.FileNotFound, $"Requested file '{message.FilePath}' does not exists.");
 				}
 
+				var effectiveUpdate = WaitForFileUpdated(message.FilePath);
 				File.Delete(message.FilePath);
+				await effectiveUpdate;
+
 				return (FileUpdateResult.Success, null);
 			}
+
+			static async Task WaitForFileUpdated(string path)
+			{
+				var file = new FileInfo(path);
+				var dir = file.Directory;
+				while (dir is { Exists: false })
+				{
+					dir = dir.Parent;
+				}
+
+				if (dir is null)
+				{
+					return;
+				}
+
+				var tcs = new TaskCompletionSource();
+				using var watcher = new FileSystemWatcher(dir.FullName);
+				watcher.Changed += async (snd, e) =>
```

Comment:
should we unsubscribe from this event at some point?

Reply:
@ajpinedam this shouldn't be required since the FileSystemWatcher is disposable. The context of watcher only exists for the duration of the WaitForFileUpdated method, and since this method blocks waiting for the TaskCompletionSource, I think we can rely on Dispose being invoked to clean up the event handler.